Detectors/ZDC/calib/src/BaselineCalibSpec.cxx

```diff
@@ -0,0 +1,147 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   BaselineCalibSpec.cxx
+/// @brief  ZDC baseline calibration
+/// @author pietro.cortese@cern.ch
+
+#include <iostream>
+#include <vector>
+#include <string>
+#include <gsl/span>
+#include "CCDB/BasicCCDBManager.h"
+#include "CCDB/CCDBTimeStampUtils.h"
+#include "CCDB/CcdbApi.h"
+#include "Framework/Logger.h"
+#include "Framework/ControlService.h"
+#include "Framework/ConfigParamRegistry.h"
+#include "Framework/CCDBParamSpec.h"
+#include "Framework/DataRefUtils.h"
+#include "DetectorsCommonDataFormats/DetID.h"
+#include "ZDCBase/ModuleConfig.h"
+#include "CommonUtils/NameConf.h"
+#include "CommonUtils/MemFileHelper.h"
+#include "CCDB/BasicCCDBManager.h"
+#include "CCDB/CCDBTimeStampUtils.h"
+#include "ZDCCalib/BaselineCalibSpec.h"
+#include "ZDCCalib/BaselineCalibData.h"
+
+using namespace o2::framework;
+
+namespace o2
+{
+namespace zdc
+{
+
+BaselineCalibSpec::BaselineCalibSpec()
+{
+  mTimer.Stop();
+  mTimer.Reset();
+}
+
+BaselineCalibSpec::BaselineCalibSpec(const int verbosity) : mVerbosity(verbosity)
+{
+  mTimer.Stop();
+  mTimer.Reset();
+}
+
+void BaselineCalibSpec::init(o2::framework::InitContext& ic)
+{
+  mVerbosity = ic.options().get<int>("verbosity-level");
+  mWorker.setVerbosity(mVerbosity);
+}
+
+void BaselineCalibSpec::updateTimeDependentParams(ProcessingContext& pc)
+{
+  // we call these methods just to trigger finaliseCCDB callback
+  pc.inputs().get<o2::zdc::ModuleConfig*>("moduleconfig");
+  pc.inputs().get<o2::zdc::BaselineCalibConfig*>("calibconfig");
+}
+
+void BaselineCalibSpec::run(ProcessingContext& pc)
+{
+  updateTimeDependentParams(pc);
+  if (!mInitialized) {
+    mInitialized = true;
+    std::string loadedConfFiles = "Loaded ZDC configuration files:";
+    {
+      // Module configuration
+      auto config = pc.inputs().get<o2::zdc::ModuleConfig*>("moduleconfig");
```

Comment:
here you are getting the CCDB objects which are already queried in the updateTimeDependentParams. It should be either / or, I would suggest to do this via updateTimeDependentParams / finaliseCCDB.